activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -489,6 +489,44 @@ def create_enum(name, values)
         exec_query(query)
       end
 
+      # Drop an existing enum type.
+      def drop_enum(name, rollback_values = nil)
+        exec_query("DROP TYPE #{name}").tap { reload_type_map }
+      end
+
+      # Rename an existing enum type to something else.
+      def rename_enum(name, options = {})
+        to = options.fetch(:to) { raise ArgumentError, ":to is required" }
+
+        exec_query("ALTER TYPE #{name} RENAME TO #{to}").tap { reload_type_map }
+      end
+
+      # Add enum value to an existing enum type.
+      def add_enum_value(type_name, value, options = {})
+        before, after = options.values_at(:before, :after)
+        sql = +"ALTER TYPE #{type_name} ADD VALUE '#{value}'"
+
+        if before && after
+          raise ArgumentError, "Cannot have both :before and :after at the same time"
+        elsif before
+          sql << " BEFORE '#{before}'"
+        elsif after
+          sql << " AFTER '#{after}'"
+        end
+
+        execute(sql).tap { reload_type_map }
+      end
+
+      # Rename enum value on an existing enum type.
+      def rename_enum_value(type_name, options = {})
```

Comment:
Renaming enum values is only supported in PostgreSQL 10 or later. We might want to raise if `postgresql_version <= 100_000`

Reply:
Good catch, thank you! I'll get this added soon.

Reply:
When you have time, would you add a condition only to allow rename enum for PostgreSQL 10 or higher?
https://www.postgresql.org/docs/10/release-10.html

> * Allow [ENUM](https://www.postgresql.org/docs/10/datatype-enum.html) values to be renamed (Dagfinn Ilmari Mannsåker)
> This uses the syntax [ALTER TYPE ... RENAME VALUE](https://www.postgresql.org/docs/10/sql-altertype.html).

Reply:
@philip-maina @yahonda this has been updated to raise an error for renaming values when the Postgres version is less than version 10.0.

I also double checked just to make sure that other enum helpers here: create, drop and other alter commands (like `RENAME` and `ADD VALUE`) were good. All other enum related commands are supported well before version 9.3 (with 9.3 being the oldest version supported by Rails) so all is good there.